### PR TITLE
[MAINTAIN-200] remove buttons margin for non IOS devices.

### PR DESCRIPTION
**Pull request by manachynskyi**
[Issue](https://openy.atlassian.net/browse/MAINTAIN-200) This PR should improve indents between buttons and the bottom of the screen on iOs/Android devices.

This PR should be compared on mobile iOS and Android (mobile resolution of desktop browser)
Steps for review (on mobile):
- Visit activity finder 4 page
- [ ] Start from `Ages` step https://i.imgur.com/1xQnb5E.png and go through all steps
- [ ] make sure on iOS we have an indent under the button (skip/next) 
- [ ] make sure on android/web indent missed and button stuck to the bottom of the screen https://i.imgur.com/0TLGSP8.png
- on the result page 
- [ ] Click on the session details
- [ ] add an item to the bookmark items https://i.imgur.com/dDX9BvZ.png
- [ ] open bookmarked items https://i.imgur.com/LmulRDz.png
- [ ] make sure the bottom region is stuck to the bottom of the screen on Android and a small indent added on iOS https://i.imgur.com/QnzIpu0.png
- [ ] open filters https://i.imgur.com/Z9xkP1e.png
- [ ] make sure buttons have same behavior as in previous cases https://i.imgur.com/t7Z0MGR.png

### CODE REVIEW

File: openy_af4_vue_app/src/main.js [hunk 1]
```diff
@@ -60,6 +61,15 @@ Vue.mixin({
...
       return ''
     }
+  },
+  computed: {
+    isIosMobile() {
+      let iosMobile = false
+      if (isMobile && isIOS) {
+        iosMobile = true
+      }
+      return iosMobile
```
Comment (hamrant) [inline openy_af4_vue_app/src/main.js]: Let's simplify:

```js
  computed: {
    isIosMobile() {
      return isMobile && isIOS
    }
  }
```

**Review by hamrant — CHANGES_REQUESTED** [review]

**Review by hamrant — APPROVED** [review]

File: openy_af4_vue_app/package.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
     "bootstrap": "^4.6.0",
     "bootstrap-vue": "^2.21.2",
     "core-js": "^3.18.3",
+    "mobile-device-detect": "^0.4.3",
```
Comment (podarok) [inline openy_af4_vue_app/package.json]: Are we sure we want to have this library https://github.com/duskload/mobile-device-detect/pulls
A lot of opened security issues @hamrant

Comment (podarok) [reply openy_af4_vue_app/package.json]: I'd rather fork it or get maintainer access and fix those security issues

**Review by podarok — CHANGES_REQUESTED** [review]
Your thoughts?

File: openy_af4_vue_app/package.json (hunk 2 above)
Comment (hamrant) [reply openy_af4_vue_app/package.json]: @podarok nice catch!

Ok, we can try a solution without an additional library with data from `Window.navigator` - https://developer.mozilla.org/en-US/docs/Web/API/Window/navigator (For now it has good browser compatibility).

How to detect iOs device - https://racase.com.np/javascript-how-to-detect-if-device-is-ios/
How to detect mobile device - https://stackoverflow.com/questions/11381673/detecting-a-mobile-browser

```js
const isMobile = navigator.userAgentData.mobile; //resolves true/false
```

cc @manachynskyi

Comment (podarok) [reply openy_af4_vue_app/package.json]: Great, let's use it instead of insecure dependency
Thank you @hamrant

**Comment by manachynskyi** [comment]
I used only checking by `userAgent`, because `userAgentData` is [experimental](https://caniuse.com/mdn-api_navigator_useragentdata) and not supported by all browsers. I tested on different iphones and ipads (on emulator) it seems that all works fine.

**Review by hamrant — APPROVED** [review]

**Review by podarok — APPROVED** [review]

**Pull request merged**